File: back_end/VehicleTracker/src/main/java/com/app/model/Checkpoint.java
package com.app.model;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@Entity
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class Checkpoint {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(unique = true , nullable = false)
    private String name;
    @Column(nullable = false)
    private double latitude;
    @Column(nullable = false)
    private double longitude;
    @Column(nullable = false)
    private double radiusKm;
}

File: back_end_dotnet/vehicleTracker_dotnet/Model/VTModels/Checkpoint.cs
using System.ComponentModel.DataAnnotations;

namespace VehicleTracker.Models;

public class Checkpoint
{
    [Key]
    public int Id { get; set; }


    [Required]
    public string Name { get; set; }


    public double Latitude { get; set; }
    public double Longitude { get; set; }
    public double RadiusKm { get; set; }
}